### PR TITLE
Add feature to change first day of the week

### PR DESCRIPTION
**Pull request by recruit-mahayash**
## Overview
Adding a feature to change first day of the week (e.g. Sunday, Monday, etc...)

## References
* [Issue #9](https://github.com/recruit-mp/LightCalendarView/issues/9)

## Technical Changes
* added the method `LightCalendarView#setFirstDayOfWeek(weekDay: WeekDay)`
* added `lcv_firstDayOfWeek` to the layout parameters
* modified `WeekDayLayout` and `DayLayout` to rearrange child views in the correct order
* added `val firstDayOfWeek: WeekDay` setting to `CalendarSettings`

## Remaining Tasks
* [x] update README.md

## Screenshot
screenshot_1481423037

## Reviewer
* @amyu

### CODE REVIEW

File: library/src/main/kotlin/jp/co/recruit_mp/android/lightcalendarview/WeekDay.kt [hunk 1]
```diff
@@ -34,9 +34,18 @@ enum class WeekDay {
     companion object {
         val ordinals = values()
         fun fromOrdinal(ordinal: Int) = ordinals[ordinal]
+
+        /**
+         * Returns the rearranged WeekDay values by sliding elements by n times
```
Comment (nobuoka) [inline library/src/main/kotlin/jp/co/recruit_mp/android/lightcalendarview/WeekDay.kt]: [コメント][このままで良い] 「sliding elements」 の部分は 「circular shift」 って言ってもいいかもしれないですね。 

* 参考 : https://en.wikipedia.org/wiki/Circular_shift

**Comment by TKolbusz** [comment]
It's buggy, January 2017 has no day 1, because it should be inflated in separate row

**Comment by TKolbusz** [comment]
I fixed issue with January 2017, I had to use Joda Time to fix that, because Java Calendar is simply poor to do that.

Changes I had to make to your feature
add JodaTime to gradle
    `compile 'net.danlew:android.joda:2.9.5.1'`

init JodaTime in LightCalendarView.kt

```
    constructor(context: Context) : this(context, null)
    {
        JodaTimeAndroid.init(getContext())
    }
    constructor(context: Context, attrs: AttributeSet) : this(context, attrs, 0)
    {
        JodaTimeAndroid.init(getContext())
    }
    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int) : this(context, attrs, defStyleAttr, 0)
    {
        JodaTimeAndroid.init(getContext())
    }
```

changed updateLayout() and added getDateTimeConstantDayOfTheWeek() in DayLayout.kt
```
    private fun updateLayout()
    {
        if (dayOfWeekOffset != settings.dayOfWeekOffset)
        {
            dayOfWeekOffset = settings.dayOfWeekOffset

            // calculate the date of top-left cell
            val cal: Calendar = Calendar.getInstance().apply {
                firstDayOfWeek = dayOfWeekOffset + 1
                val localDate = LocalDate(month.time).dayOfMonth().withMinimumValue().withDayOfWeek(getDateTimeConstantDayOfTheWeek())
                timeInMillis = localDate.toDate().time
            }
            firstDate = cal

            // remove all children
            removeAllViews()

            // populate children
            populateViews()
        }
    }

    private fun getDateTimeConstantDayOfTheWeek(): Int
    {
        when (dayOfWeekOffset)
        {
            1 -> return DateTimeConstants.MONDAY
            2 -> return DateTimeConstants.TUESDAY
            3 -> return DateTimeConstants.WEDNESDAY
            4 -> return DateTimeConstants.THURSDAY
            5 -> return DateTimeConstants.FRIDAY
            6 -> return DateTimeConstants.SATURDAY
            else ->
            {
                return DateTimeConstants.SUNDAY
            }
        }
    }


```

**Comment by recruit-mahayash** [comment]
@TKolbusz,

Thank you very much for the comment! We actually solved the issue with a different approach, but your investigation and contribution is highly appreciated. If you have any customizations or bug fixes you made, please feel free to fork this repository to your account, commit and push there, and then create a pull request for that change.

Thanks!

**Comment by TKolbusz** [comment]
@recruit-mahayash I customized your lib to my needs, I needed to dynamically change all colors at runtime and make only one date selectable at a time. Those are static changes, it would be nice if you add option to select only one day at a time, and add highly customizable coloring, setCurrentDayTextColor(int color) for example. My changes are not suitable for commit because I removed some of the features, different color of current day for example.

**Pull request merged**